Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp

```diff
@@ -156,6 +164,16 @@ void RemoteRenderingBackendProxy::disconnectGPUProcess()
     m_didRenderingUpdateID = { };
     m_streamConnection->invalidate();
     m_streamConnection = nullptr;
+    m_gpuProcessConnection = nullptr;
+}
+
+void RemoteRenderingBackendProxy::notifyMessageDidTimeout()
+{
+    ensureOnMainRunLoop([weakGPUProcessConnection = m_gpuProcessConnection]() {
```

Comment:
The intention was for this to use the same member as you're using in [#29173 ](https://github.com/WebKit/WebKit/pull/29173), so we can't move it out as its used later on.

I don't think we need to invalidate it, the expectation is that the GPU process will 'crash' in the near future and we'd take the usual restart path at that point. The calling code here will just attempt to carry on (and probably immediately block in the next message send attempt).

Reply:
> as its used later on.

But for what is it used later on?
When the message send to GPUP times out, the GPUP is defunct.It cannot be trusted and there's nothing to be done with it.

Also we cannot let anybody send any more messages, as the buffer is in unknown state (it will assert after timeout)..

Yes, at the moment invalidation would actually be problematic, as that will disconnect the didClose handler that this commit depends on..